packages/runtime/container-runtime/src/deltaManagerRuntimeProxy.ts

```diff
@@ -0,0 +1,127 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+	IDeltaManager,
+	IDeltaManagerEvents,
+	IDeltaQueue,
+	IDeltaSender,
+	ReadOnlyInfo,
+} from "@fluidframework/container-definitions";
+import { EventForwarder } from "@fluidframework/common-utils";
+import {
+	IClientConfiguration,
+	IClientDetails,
+	IDocumentMessage,
+	ISequencedDocumentMessage,
+	ISignalMessage,
+} from "@fluidframework/protocol-definitions";
+import { summarizerClientType } from "./summary";
+
+/**
+ * Proxy to the real IDeltaManager for the container runtime. It is used to restrict certain access to the runtime and
+ * layers below it.
+ */
+export class DeltaManagerRuntimeProxy
```

Comment:
We need to give it better name. If you think it will have one responsibility, then I'd call it according to this responsibility. 
I think it's better to have many adapters like that in the future, each with its own responsibility (then merging all into one class), so I think we should rename this class / file accordingly.

Reply:
Updated the name to `DeltaManagerSummarizerProxy`